### ISSUE TITLE
Scrollarea for commit messages

### ISSUE THREAD

**Issue by ASMfreaK** [nicoespeon/gitgraph.js#138]
It would be nice if long commit messges could be scrolled. IDK if it can be implemented, though.

**Comment by nicoespeon** [comment]
Maybe instead of scrolling we could make it fit in a defined "box".

I have a question though: are you speaking of commit messages displayed in tooltips in compact (and horizontal) mode or regular ones displayed in vertical mode? Or both?

**Comment by nicoespeon** [comment]
So, this is technically _not_ resolved for the `gitgraph.js` package… But we released `@gitgraph/react` in v1.0.0-0 and this issue was fixed here. You can customize how commit message is rendered, thus insert whatever you want:

![image](https://user-images.githubusercontent.com/1094774/52529123-5b5ef600-2cba-11e9-8053-d86c47cb045c.png)

```jsx
const renderMessage: CommitOptions["renderMessage"] = (
  commit,
  commitMessageX,
) => {
  return (
    <g
      transform={`translate(${commitMessageX - commit.x}, ${
        commit.style.dot.size
      })`}
    >
      <text fill={commit.style.dot.color}>
        {commit.hashAbbrev} - {commit.subject}
      </text>
      <foreignObject width="600" x="10">
        <p>
          My money's in that office, right? If she start giving me some
          bullshit about it ain't there, and we got to go someplace else and
          get it, I'm gonna shoot you in the head then and there. Then I'm
          gonna shoot that bitch in the kneecaps, find out where my goddamn
          money is. She gonna tell me too. Hey, look at me when I'm talking
          to you, motherfucker. You listen: we go in there, and that nigga
          Winston or anybody else is in there, you the first motherfucker to
          get shot. You understand?
        </p>
      </foreignObject>
    </g>
  );

return (
  <Gitgraph>
    {(gitgraph) => {
      gitgraph
        .commit({ subject: "Initial commit" })
        .commit({ subject: "Another commit" })
        .commit({
          subject: "Do something crazy",
          renderMessage,
        });

      gitgraph
        .branch("dev")
        .commit({
          subject: "Oh my god",
        })
        .commit({
          subject: "This is a saxo!",
        });
    }}
  </Gitgraph>
);
```

Thus, I'm closing this issue as we won't maintain the `gitgraph.js` package. There will be a `@gitgraph/js` package soon to replace it. 

We still accept PRs if someone really needs it, but I'd suggest moving to the aforementioned packages. Thanks 🙏
